python/paddle/fluid/dygraph/base.py

```diff
@@ -64,17 +64,35 @@ def program_desc_tracing_guard(enable):
 
 @signature_safe_contextmanager
 def param_guard(parameters):
-    # Note: parameters is a reference of self._parameters
+    # Note: parameters is a reference of self._parameters or self._buffers
     if not framework.in_dygraph_mode() and parameters:
         origin_parameters = parameters.copy()
         for name, var_base in parameters.items():
             if isinstance(var_base, core.VarBase):
-                new_var = framework.Parameter(
-                    var_base.block,
-                    var_base.shape,
-                    var_base.dtype,
-                    var_base.type,
-                    name=var_base.name)
+                # Note(Aurelius84): The var type in self._parameters is ParamBase,
+                # but VarBase in self._buffers. In dy2static, we convert all the VarBases from
+                # `self._parameters` and `self._buffers` into framework.Parameter.
+                # Because users can create a VarBase in `__init__` equivalent to a ParamBase like a
+                # `mask` Tensor or `hidden_0` in RNN layers, which is necessary for inferring.
+                # So we convert and save them into inference model.
+                if isinstance(var_base, framework.ParamBase):
+                    new_var = var_base._to_static_var(to_parameter=True)
+                else:
+                    # A VarBase in self._buffers may be passed to initialize multiple sub_layers.
+                    is_created = var_base.name in var_base.block.vars
```

Comment:
Seems `is_created` is used only once, you can inline the line 82 into line 83, such as

if var_base.name in var_base.block.vars

Reply:
Thanks，done.